src/test/ui/consts/static_mut_containing_mut_ref2.mut_refs.stderr

```diff
@@ -1,9 +1,9 @@
-error[E0764]: mutable references are not allowed in statics
-  --> $DIR/static_mut_containing_mut_ref2.rs:7:46
+error[E0080]: could not evaluate static initializer
+  --> $DIR/static_mut_containing_mut_ref2.rs:7:45
    |
 LL | pub static mut STDERR_BUFFER: () = unsafe { *(&mut STDERR_BUFFER_SPACE) = 42; };
-   |                                              ^^^^^^^^^^^^^^^^^^^^^^^^^^ `&mut` is only allowed in `const fn`
+   |                                             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ modifying a static's initial value from another static's initializer
```

Comment:
This is bad. We no longer catch this pre-monomorphization even though it has`&mut`. You'll need to handle this case somehow.

Reply:
I tried, but I see no way to distinguish this from the `&mut STDERR_BUFFER_SPACE` that we'd see for just `let x = STDERR_BUFFER_SPACE` without going all the way to having custom `Rvalue`s for that

Reply:
There's always the HIR const-checker I guess :laughing:. I haven't got around to removing it yet.

As written, this PR allows stuff like `let x = &mut STATIC_MUT` in a const-context on stable, so you'll need to handle it somehow.

Reply:
Unfortunately

```rust
struct Foo {
    field: *mut i32
}

static mut FOO: Foo = Foo {
    field: &mut [42] as *mut [i32] as *mut i32,
};
```

is stable, so I can't just ban mutable references.

Reply:
But yea, I gottaa figure something out how to not expose it even more...